apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -272,6 +272,9 @@ type ClusterQueuePreemption struct {
 	// +kubebuilder:default=Never
 	// +kubebuilder:validation:Enum=Never;LowerPriority
 	WithinClusterQueue PreemptionPolicy `json:"withinClusterQueue,omitempty"`
+
+	// +kubebuilder:default="flase"
+	PreemptBasedOnFlavorOrder bool `json:"preemptBasedOnFlavorOrder"`
```

Comment:
In general, booleans might not be the best API https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

For example:

```yaml
  flavorFungibility: NextFlavorBeforePreemption | PreemptionBeforeNexFlavor
```

I don't really like the names I'm proposing, but I'll leave it to you to come up with other ideas.

Reply:
how about 
```
PreemptionTiming: BeforeNextFlavor | AfterAllFlavors
```